#StartFloat syntax

cbmarini opened the pull request:
This PR updates the `#StartFloat` syntax as discussed in #676. The new syntax is
```
#StartFloat <precision>, MZV = <weight>
``` 
where the precision can either be in bits (`b`) or digits (`d`). For example:
```
#StartFloat 50b, MZV = 2
#StartFloat 20d, MZV = 3
```
Notes:
- To properly test whether the correct number of digits are printed, PR #676 should be merged first.
- In this regard, it is also worth reviewing whether the two separate calls to `SetFloatPrecision` from `SetupMZVTables` and `SetupMPFTables` are both needed (both increase the precision by the MZV weight). Likely, a single call in e.g. `DoStartFloat` should be sufficient.

jodavies commented:
This looks like a good change, but remember to also update the manual.

I would perhaps also merge the commits, since the CI will not pass on the first commit alone, and perhaps makes bisection a bit less smooth.

>     * In this regard, it is also worth reviewing whether the two separate calls to `SetFloatPrecision` from `SetupMZVTables` and `SetupMPFTables` are both needed (both increase the precision by the MZV weight). Likely, a single call in e.g. `DoStartFloat` should be sufficient.

I think this is also a good idea, but maybe is better as part of #676 which is already changing how the allocation and precision is handled.

jodavies commented:
I would say that the order need not be relaxed, it is OK to require the precision first.

Making the precision mandatory, as in the manual, so that the user is forced to choose a sensible value for their requirements would also be fine for me.

Could you add a test (or edit a couple of the existing ones) to use the "digits" version of the precision?

cbmarini commented:
Yes, I will add/modify tests after PR #676 is merged.

cbmarini commented:
I’ve now implemented the new syntax for `Format floatprecision <precision>(b/d)`, matching the syntax used in `#StartFloat`.

In addition, I added proper scientific notation output for Mathematica and a corresponding test.
Could someone confirm whether the Maple syntax is correct?
Also, would it make sense to include Fortran output in this test as well?

cbmarini commented on doc/manual/float.tex:
```diff
@@ -12,10 +12,12 @@ \chapter{Floating point}
+be a positive integer followed by either \texttt{b} (for precision in bits) 
+or \texttt{d} (for precision in decimal digits).
+\FORM{} will round to at least this precision. Because the internal 
 routines work with WORDs the precision will be rounded up to the nearest 
```
Ah yes. Maybe we should add something on the raw form of the floats, i.e. the `float_` function? Otherwise leave this sentence out?
